test/logs/helpers/taskrun_logs.go

```diff
@@ -0,0 +1,164 @@
+/*
+Copyright 2017 Google Inc. All Rights Reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logs
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+
+	pipelinev1alpha1 "github.com/knative/build-pipeline/pkg/client/clientset/versioned/typed/pipeline/v1alpha1"
+	duckv1alpha1 "github.com/knative/pkg/apis/duck/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+)
+
+func TaskrunLogs(ctx context.Context, cfg *rest.Config, trName, namespace string, out io.Writer) error {
+	pclient, err := pipelinev1alpha1.NewForConfig(cfg)
+	if err != nil {
+		return err
+	}
+	tr, err := pclient.TaskRuns(namespace).Get(trName, metav1.GetOptions{IncludeUninitialized: true})
+	if err != nil {
+		return err
+	}
+	if tr != nil {
+		trCond := tr.Status.GetCondition(duckv1alpha1.ConditionSucceeded)
+		if trCond.IsFalse() {
+			return fmt.Errorf("taskrun failed with msg: %v", trCond.Message)
+		}
+	}
+
+	client, err := corev1.NewForConfig(cfg)
+	if err != nil {
+		return fmt.Errorf("getting corev1 kubernetes client: %v", err)
+	}
+
+	podName := tr.Status.PodName
+	pods := client.Pods(namespace)
+	watcher := PodWatcher{
+		Pods: pods,
+		Name: podName,
+	}
+	if err := watcher.Start(ctx); err != nil {
+		return fmt.Errorf("watching pod: %v", err)
+	}
+
+	pod, err := watcher.WaitForPod(ctx, func(p *v1.Pod) bool {
+		return len(p.Status.InitContainerStatuses) > 0
+	})
+	if err != nil {
+		return err
+	}
+
+	for i, container := range pod.Status.InitContainerStatuses {
+		pod, err := watcher.WaitForPod(ctx, func(p *v1.Pod) bool {
+			waiting := p.Status.InitContainerStatuses[i].State.Waiting
+			if waiting == nil {
+				return true
+			}
+
+			if waiting.Message != "" {
+				fmt.Fprintln(out, Red(fmt.Sprintf("[%s] %s", container.Name, waiting.Message)))
+			}
+
+			return false
+		})
+		if err != nil {
+			return fmt.Errorf("waiting for container: %v", err)
+		}
+
+		container := pod.Status.InitContainerStatuses[i]
```

Comment:
hm we'll have to remember to update this once https://github.com/knative/build/pull/470 lands!

since landing https://github.com/knative/build/pull/470 could break this tool, what do you think about adding some kind of super basic sanity check test that will verify the tool keeps working? otherwise we could easily forget about it (if we're not using it regularly, which I bet we will be!)

Reply:
i also wonder if the shifting implementation could make it a good idea to abstract the init container based portion of this logic into a function or package that we can swap out when we land https://github.com/knative/build/pull/470

Reply:
I think this should work after build PR lands. TaskRun status hold podname and that is used for tracking pod logs.